### PR TITLE
Using all cores for make check | workaround for make check failure for libevent

### PR DESCRIPTION
**Pull request by jumde**
Fix https://github.com/brave/tor_build_scripts/issues/32
Fix https://github.com/brave/tor_build_scripts/issues/33

1. Use all cores for running tests
2. Using test/regress.c from a local repo because make check hangs in libevent. See: https://github.com/libevent/libevent/issues/747 for more details. Updated test/regress.cc disables: del_wait | immediatesignal | signal_switchbase | signal_while_processing tests

### CODE REVIEW

File: build_darwin.sh [hunk 1]
```diff
@@ -17,11 +19,17 @@ curl -fsSL "https://github.com/libevent/libevent/releases/download/release-$LIBE
...
+# See: https://github.com/libevent/libevent/issues/747 for more details
+# Updated test/regress.cc disables: del_wait | immediatesignal | 
+# signal_switchbase | signal_while_processing tests
+
 gpg --keyserver "$KEYSERVER" --recv-keys $LIBEVENT_KEY && \
 gpg libevent-$LIBEVENT_VERSION.tar.gz.asc && \
 echo "$LIBEVENT_HASH  libevent-$LIBEVENT_VERSION.tar.gz" | shasum -a 256 -c - && \
 tar -zxvf libevent-$LIBEVENT_VERSION.tar.gz && \
 cd libevent-$LIBEVENT_VERSION && \
+curl -o test/regress.c https://raw.githubusercontent.com/jumde/libevent/legacy_fork_failure/test/regress.c && \
```
Comment (jumde) [inline build_darwin.sh]: Issue to revert this change once the upstream change is fixed: https://github.com/brave/tor_build_scripts/issues/34

File: build_darwin.sh [hunk 2]
```diff
@@ -1,12 +1,14 @@
 #!/bin/sh
 set -eu
 
+jobs=$(sysctl -n hw.logicalcpu_max)
+
```
Comment (riastradh-brave) [inline build_darwin.sh]: Can this be taken as an argument instead, and only fall back to hw.logicalcpu_max as a default?  (Generally I try to avoid hard-coding logic to unconditionally use all CPUs: if you're trying to do multiple things at a time, you might want to limit them to smaller subsets of the CPUs rather than have them competing with each other -- and the web browser you're trying to use to read Twitter while you run your builds -- over all the CPUs.)

File: build_darwin.sh (hunk 1 above)
Comment (riastradh-brave) [reply build_darwin.sh]: Can we store this file in the repository?  If not, let's at least commit a hash of the file and verify it before using it.

File: build_darwin.sh (hunk 2 above)
Comment (riastradh-brave) [reply build_darwin.sh]: [not blocking; would be nice]

File: build_darwin.sh [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
+  jobs=$(sysctl -n hw.logicalcpu_max)
+fi
 
 curl -fsSL "https://www.openssl.org/source/openssl-$OPENSSL_VERSION.tar.gz" -o openssl-$OPENSSL_VERSION.tar.gz && \
 echo "$OPENSSL_HASH  openssl-$OPENSSL_VERSION.tar.gz" | shasum -a 256 -c - && \
 tar -xvzf openssl-$OPENSSL_VERSION.tar.gz && \
 cd openssl-$OPENSSL_VERSION && \
 ./config --prefix=$PWD/install darwin64-x86_64-cc no-shared no-dso && \
-make
+make ${jobs:+-j${jobs}}
```
Comment (riastradh-brave) [inline build_darwin.sh]: Nit: No need for the `${jobs:+...}` part if `jobs` is always defined.
